src/DIRAC/Resources/IdProvider/Utilities.py

```diff
@@ -3,78 +3,62 @@
+            providers.append(name)
+    return S_OK(providers)
 
 
-def getSettingsNamesForClientID(clientID):
-    """Get identity providers for clientID
+def getIdProviderIdentifierFromIssuerAndClientID(issuer, clientID):
+    """Get identity provider for issuer
 
-    :param str clientID: clientID
+    :param str issuer: issuer
+    :param str clientID: client ID
 
-    :return: S_OK(list)/S_ERROR()
+    :return: S_OK(str)/S_ERROR()
     """
-    names = []
-    result = getProvidersForInstance("Id")
+    result = getIdProviderIdentifiers()
     if not result["OK"]:
         return result
-    for name in result["Value"]:
-        res = gConfig.getValue(f"/Resources/IdProviders/{name}/client_id")
-        if res and clientID == res:
-            names.append(name)
-    return S_OK(names) if names else S_ERROR(f"Not found provider with {clientID} clientID.")
-
-
-def getProvidersForInstance(instance, providerType=None):
-    """Get providers for instance
-
-    :param str instance: instance of what this providers
-    :param str providerType: provider type
 
-    :return: S_OK(list)/S_ERROR()
-    """
-    providers = []
-    instance = f"{instance}Providers"
-    result = gConfig.getSections(f"/Resources/{instance}")
+    for identifier in result["Value"]:
+        testedClientID = gConfig.getValue(f"/Resources/IdProviders/{identifier}/client_id")
+        if testedClientID and testedClientID == clientID:
+            # Found the client ID but need to check the issuer
+            # 2 different issuers could theoretically have a same client ID
+            testedIssuer = gConfig.getValue(f"/Resources/IdProviders/{identifier}/issuer")
 
-    # Return an empty list if the section does not exist
-    if not result["OK"] or not result["Value"] or not providerType:
-        return result
+            if testedIssuer and testedIssuer.strip("/") == issuer.strip("/"):
+                return S_OK(identifier)
 
-    for prov in result["Value"]:
-        if providerType == gConfig.getValue(f"/Resources/{instance}/{prov}/ProviderType"):
-            providers.append(prov)
-    return S_OK(providers)
+    return S_ERROR("IdProvider not found")
 
 
-def getProviderInfo(provider):
+def getIdProviderParameters(idProvider):
     """Get provider info
 
     :param str provider: provider
 
     :return: S_OK(dict)/S_ERROR()
     """
-    result = gConfig.getSections("/Resources")
+    result = gConfig.getOptionsDict(f"/Resources/IdProviders/{idProvider}")
     if not result["OK"]:
         return result
-    for section in result["Value"]:
-        if section.endswith("Providers"):
-            result = getProvidersForInstance(section[:-9])
-            if not result["OK"]:
-                return result
-            if provider in result["Value"]:
-                return gConfig.getOptionsDictRecursively(f"/Resources/{section}/{provider}/")
-    return S_ERROR(f"{provider} provider not found.")
+
+    return S_OK(result["Value"])
```

Comment:
I think the block from line 60 is equivalent to do

```
return gConfig.getOptionsDict(f"/Resources/IdProviders/{idProvider}")
```

Reply:
Indeed, I am going to fix this.
Do we even need this function then? (`getIdProviderParameters`)

Reply:
It might not be needed indeed.